packages/@n8n/nodes-langchain/nodes/agents/Agent/agents/ToolsAgent/error-handling.ts

```diff
@@ -0,0 +1,7 @@
+export function getErrorMessageByStatusCode(statusCode: number): string | undefined {
+	const errorMessages: Record<number, string> = {
+		404: 'The resource was not found. Please check the model name',
+	};
+
+	return errorMessages[statusCode];
+}
```

Comment:
The Google Vertex models could also be used with other agents/chains. It would be nice to externalize these errors and use it everywhere where there's model input

Reply:
Agree. Moved error handling to the Vertex model node